File: src/Core/ViewModels/SendMessageViewModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Core.ViewModels
{
    public sealed class SendMessageViewModel
    {
        [Required]
        public string RecipientUserId { get; set; }

        [Required]
        public string Context { get; set; }

        [Required]
        public DateTime DateSent { get; set; }
    }
}

File: src/AngularApp/ClientApp/src/app/models/SendMessageViewModel.ts
export class SendMessageViewModel {
  constructor(context: string, recipientUserId: string) {
    this.context = context;
    this.dateSent = new Date();
    this.recipientUserId = recipientUserId;
  }

  public context: string;
  
  public dateSent: Date;
  
  public recipientUserId: string; 
}